### DISCUSSION TITLE
Display MappedValue from uiFixedSliderContent on a uiLabel

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#275] [Q&A] [answered]
Hi everyone,

I struggle to find a way to display a fixed value on a label.
Of course we all know getMappedValue() doesn't work unfortunately.

I tried several option like:

```
modMappedValue = mod:getComponent():getMappedValue()
modMappedValue = mod:getProperty("modulatorMappedValue")
modMappedValue = mod:getComponent():getProperty("modulatorMappedValue")
```

I had no success.

Since I have all the values available in the uiFixedSliderContent and I can get the modulator current value There should be a way to get the fixed value but I did not figured out it yet.

Do you know if there's a workaround so that we could get the current fixed value from the uiFixedSliderContent ?

Thanks a lot. Have an excellent weekend

Damien

**Comment by Godlike-Productions** [comment]
You'll need to use getValueMapped(), though there seems to be a bug with it, in that it will return the mapped index and not the displayed value.

For example if your value list is
```
5=1
10=2
15=5
```

and you have the slider set to the max value then getValueMapped will return 5. getValue() will return 2 (value is always mapped starting at 0).

You'll then need a lookup table like this.

```
byFivesLookup = {[1]=5, [2]=10,[5]=15}
```

You can return the displayed value like this

```
mod1 = panel:getModulatorByName("mod1")
modMappedValue = byFivesLookup[mod1:getValueMapped()]
```

**Comment by damiensellier** [comment, marked as answer]
Alright I made the most simple function to finally get the fixed/mapped value of a uiFixedSlider and uiImageFixedSlider.
Beforhand, I have my fixed values stored like this exemple, (values are separated by \n (LUA new line tag) to fit the uiFixedSliderContent box properly): 

`MainFxParam13:getComponent():setProperty("uiFixedSliderContent","Off\nC1\nC#1\nD1\nD#1\nE1\nF1\nF#1\nG1\nG#1\nA1\nA#1\nB1\nC2\nC#2\nD2\nD#2\nE2\nF2\nF#2\nG2\nG#2\nA2\nA#2\nB2\nC3\nC#3\nD3\nD#3\nE3\nF3\nF#3\nG3\nG#3\nA3\nA#3\nB3\nC4\nC#4\nD4\nD#4\nE4\nF4\nF#4\nG4\nG#4\nA4\nA#4\nB4\nC5\nC#5\nD5\nD#5\nE5\nF5\nF#5\nG5\nG#5\nA5\nA#5\nB5\nC6", true)`


Here is the following script:

```
function split(pString, pPattern)
   local Table = {}  -- NOTE: use {n = 0} in Lua-5.0
   local fpat = "(.-)" .. pPattern
   local last_end = 1
   local s, e, cap = pString:find(fpat, 1)
   while s do
      if s ~= 1 or cap ~= "" then
     table.insert(Table,cap)
      end
      last_end = e+1
      s, e, cap = pString:find(fpat, last_end)
   end
   if last_end <= #pString then
      cap = pString:sub(last_end)
      table.insert(Table, cap)
   end
   return Table
end
```


```
modFixedSliderContentTable = split(mod:getComponent():getProperty("uiFixedSliderContent"),'\n') -- create table from the string, values were separated by LUA "New Line" tag  \n . Note LUA arrays start at 1)
modMappedValue = modFixedSliderContentTable[value+1] -- modValue starts at 0 hence +1
panel:getLabel("lblValueChange"):setText(modMappedValue) -- displays the fixedValue in the dedicated uiLabel
```

And it's all good!


Enjoy your weekend ;-)
